src/Microsoft.VisualStudio.Editors/ResourceEditor/Resource.vb

```diff
@@ -30,6 +30,7 @@ Namespace Microsoft.VisualStudio.Editors.ResourceEditor
     ''' </remarks>
     <Serializable>
     <TypeDescriptionProvider(GetType(ResourceTypeDescriptionProvider))>
+    <KnownType(GetType(ResXDataNode))>
```

Comment:
This is the only thing I needed to add to any of the types being serialized. It seems that DataContractSerializer is slightly (very slightly) more strict and needs to know ahead of time that it will d/encode this particular type. All other types used did not need any additional information. This is why I said that DataContractSerialize will cover **99%** of the types used without any changes.

Reply:
Interesting. Have you checked other common types, like `Bitmap` and `Icon`?

Reply:
I used Bitmap as a primary testing scenario. In older implementations, I tried Icon, audio files, and other files. I need to test them again with this implementation. But, my assumption has been if Bitmap works, the other types will too.

Reply:
Please test those other common types before merging this.

Reply:
I added tests with some types. However, the top-level 'common' types (`Resource`, `MultipleValuesStore`) for this serialization implementation cannot be done as they require the context of Visual Studio to create. They have quite deep hierarchies of objects. I think mocking them would take longer than it is worth, considering this is legacy code that has existed this long without any testing.